File: Go/ParkingLot2-go/ticket.go
package main

import (
	"fmt"
	"time"
)

type Ticket struct {
	Floor *Floor
	Slot  *Slot2
	Time  int64
}

func NewTicket(floor *Floor, slot *Slot2) (*Ticket, error) {
	if floor == nil || slot == nil {
		return nil, fmt.Errorf("invalid ticket details!")
	}
	return &Ticket{Floor: floor, Slot: slot, Time: time.Now().UnixMilli()}, nil
}

func (t *Ticket) GetInfo() (floorID, slotID int, ticketTime int64) {
	return t.Floor.FloorNum, t.Slot.ID, t.Time
}

File: JavaScript/ParkingLot2/Ticket.js
class Ticket {
  floor = 0;
  slot = 0;
  time = new Date();

  constructor(floor, slot) {
    if ([floor, slot].some(ele => !ele)) {
console.log(floor, slot);
      throw new Error('Invalid ticket details!');
    }

    this.floor = floor;
    this.slot = slot;
    this.time = Date.now();
  }

  getInfo() {
    return {
      floorId: this.floor,
      slotId: this.slot,
      time: this.time
    };
  }
}

module.exports = { Ticket };
